datafusion/execution/src/config.rs

```diff
@@ -157,30 +157,34 @@ impl SessionConfig {
     }
 
     /// Set a configuration option
-    pub fn set(mut self, key: &str, value: ScalarValue) -> Self {
-        self.options.set(key, &value.to_string()).unwrap();
-        self
+    pub fn set(self, key: &str, value: ScalarValue) -> Self {
```

Comment:
I would still let this method to insert into options, but since we have type specific methods, this method can probably be a helper and be not public

Reply:
Okay, will rework it.

Reply:
I think it still might be helpful to have `fn set` -- how about we leave it in this PR and we can propose a follow on PR to remove it.